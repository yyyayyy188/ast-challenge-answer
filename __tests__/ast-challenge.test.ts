import actHookCreate from "../src/index";
import generator from "@babel/generator";
import example from "../example-methods.json";
import * as t from "@babel/types";

test("create returns correct AST node", () => {
  //   console.log(example);
  const astJson = {
    Pools: {
      requestType: "QueryPoolsRequest",
      responseType: "QueryPoolsResponse",
    },
  };
  const result: any = actHookCreate(astJson);
  console.log(generator(result).code);
  expect(generator(result).code).toMatchSnapshot();
});
