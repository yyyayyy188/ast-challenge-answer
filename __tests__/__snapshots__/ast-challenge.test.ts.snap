// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create returns correct AST node 1`] = `
"export interface UsePoolsQuery<TData> extends ReactQueryParams<QueryPoolsResponse, TData> {
  request?: QueryPoolsRequest;
}

const usePools = <TData = QueryPoolsResponse,>({
  request,
  options
}): UseQueryResult<TData> => {
  return useQuery([\\"PoolsQuery\\", request], () => {
    if (!queryService) throw new Error(\\"Query Service not initialized\\");
    return queryService.pools(request);
  }, options);
};"
`;
